units/src/amount.rs

```diff
@@ -1344,10 +1344,10 @@ impl FromStr for SignedAmount {
 }
 
 impl TryFrom<Amount> for SignedAmount {
-    type Error = ParseAmountError;
+    type Error = OutOfRangeError;
 
     fn try_from(value: Amount) -> Result<Self, Self::Error> {
-        value.to_signed()
+        value.to_signed().map_err(|_| OutOfRangeError::too_big(true))
```

Comment:
Oh, `to_signed` should return the same error type.

Reply:
It doesn't [`to_signed()`](https://github.com/rust-bitcoin/rust-bitcoin/blob/53461f71c9cd6ed49acff61396758f53f8fa3de8/units/src/amount.rs#L867C1-L874C6)

Reply:
I know, we should change it. Could be in this PR (ideally as the first commit) or in another one.

Reply:
I have squashed all the commits into one